core/common/src/main/java/alluxio/Configuration.java

```diff
@@ -406,10 +407,36 @@ public static Duration getDuration(PropertyKey key) {
   }
 
   /**
-   * @return a view of the internal {@link Properties} of as an immutable map
+   * @return a map of the resolved properties represented by this configuration,
+   * including all default properties
    */
   public static Map<String, String> toMap() {
-    return Collections.unmodifiableMap(PROPERTIES);
+    Map<String, String> map = toRawMap();
```

Comment:
We still need to check if the map already contains the keyName, because some default keys might have already been set in the PROPERTIES and we do not want to override them.

Besides, we still need to check if the defaultValue is null. if it is null, the get(key) will throw undefined configuration key exception (No value set for configuration key fs.swift.user).

get(key) do not help the logic become simpler.
``` java
    for (PropertyKey key : PropertyKey.defaultKeys()) {
      String keyName = key.getName();
      String defaultValue = key.getDefaultValue();
      if (!map.containsKey(keyName) && defaultValue != null) { // cannot delete
        map.put(keyName, defaultValue); // change to get(keyName) is wasteful
      }
    }
```

Reply:
Change to:
```
  public static Map<String, String> toMap() {
    Map<String, String> map = toRawMap();
    for (Map.Entry<String, String> entry : map.entrySet()) {
      String value = entry.getValue();
      if (value != null) {
        map.put(entry.getKey(), lookup(value));
      }
    }
    return map;
  }
```
because we need to check if the value is null. If the value is null, the lookup with throw exception.